config/src/main/java/org/springframework/security/config/annotation/web/builders/HttpSecurity.java

```diff
@@ -1026,6 +1026,26 @@ public HttpSecurity addFilter(Filter filter) {
 	}
 
 	/**
+	 * Adds the Filter at the location of the specified Filter class. For example, if you
+	 * want the filter CustomFilter to be registered in the same position as
+	 * {@link UsernamePasswordAuthenticationFilter}, you can invoke:
+	 *
+	 * <pre>
+	 * addFilterAt(new CustomFilter(), UsernamePasswordAuthenticationFilter.class)
+	 * </pre>
+	 *
+	 * @param filter the Filter to register
+	 * @param atFilter the location of another {@link Filter} that is already registered
+	 * (i.e. known) with Spring Security.
+	 * @return the {@link HttpSecurity} for further customizations
+	 */
+	public HttpSecurity addFilterAt(Filter filter, Class<? extends Filter> atFilter) {
+		this.comparitor.registerAt(filter.getClass(), atFilter);
+		this.filters.add(filter);
```

Comment:
Both addFilterBefore() and addFilterAfter() call the addFilter() method to add the filter which also does a pre-check before adding. However, the new addFilterAt() method directly puts the filter in the filters member variable avoiding the extra check in addFilter(). Although the first call to comparitor.registerAt() (if it succeeds) would ensure that the addFilter check will succeed too. So the current implementation would work the same way. But it's not consistent with the other 2 methods of adding. Is there a reasoning behind this that I'm not seeing?